src/lib/optimizer/strategy/between_composition_rule.cpp

```diff
@@ -348,12 +348,12 @@ void BetweenCompositionRule::_replace_predicates(const std::vector<std::shared_p
   }
 }
 
-void BetweenCompositionRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) const {
-  if (node->type == LQPNodeType::Predicate) {
+void BetweenCompositionRule::_apply_to_plan_without_subqueries(const std::shared_ptr<AbstractLQPNode>& lqp_root) const {
```

Comment:
I know that we have no great definition of "root", but given that we call `_apply_to_plan_without_subqueries_inputs`, this is not really the root, is it?

If this is only about clang-tidy complaining, we could use `node` and add a `NOLINTNEXTLINE` with an explanation.

This applies to other rules, too.

Reply:
The function is called by `AbstractRule` with
1. the plan's root node
2. subquery plans' root nodes
3. inputs of any given node (→ `_apply_to_plan_inputs_without_subqueries`)

Also, it is recursively called from `BetweenCompositionRule::_apply_to_plan_without_subqueries`.

Reply:
Also, I chose `lqp_root` to make the variable stand out more clearly among other "node"-variables. What about `node_in` ?

Reply:
I'm talking about the legacy part. As long as that path is taken, I would consider giving it a different name at least for those rules. Once #2276 is done, `lqp_root` should be the correct name.

Reply:
Don't get me wrong. I like `lqp_root`. For now, `node_in` works for me. Alternatively, if you want to quickly address #2276 once this done, we can leave `lqp_root` for now.

Reply:
Ok, let's keep `lqp_root` then. Still fresh, so I'll try to address #2276 next  ;)